docs/hugo/content/ADR-2023-03-Post-Reconciliation-Extension.md

```diff
@@ -0,0 +1,46 @@
+---
+title: '2023-03: Post Reconciliation Extensions'
+---
+
+## Context
+
+The current behaviour of Azure Service Operator (ASO) is to set the `Ready` status on a resource by default after a successful creation of a resource. 
+
+For some resources, this is not ideal as we need to check the status of a created resource post reconciliation and set a condition accordingly. 
```

Comment:
Remember the extension point is generic - while our current use is for `PrivateEndpoint`, we're not designing it to handle that case only. The design and the description need to allow for many uses - the defining characteristic of this extension point is _when_ it gets applied, not _which_ resources it's applied to.

```suggestion
For some resources, this is not sufficient as there may be additional steps required after initial creation of the resource. We need the ability to check the status of a created resource post reconciliation and set a condition accordingly. 
```